tiledb/sm/query/readers/result_coords.h

```diff
@@ -273,15 +275,38 @@ struct GlobalOrderResultCoords
    * @param cmp The comparator class. Calling cmp(current, next) should tell us
    * if current is bigger or equal than next in the order of the comparator.
    *
+   * @postcondition this method has no side-effects if it is never called
+   * concurrently by multiple threads. If multiple threads call this method
+   * concurrently then results may be undefined.
```

Comment:
Thanks for highlighting this. We need to be more aware that not all current reader steps are designed for concurrency and that they can silently cause undefined behavior.